src/main/java/com/amazonaws/services/neptune/io/KinesisConfig.java

```diff
@@ -23,22 +24,22 @@ public class KinesisConfig {
     private final Stream stream;
     private static final Logger logger = LoggerFactory.getLogger(KinesisConfig.class);
 
-    public KinesisConfig(String streamName, String region, LargeStreamRecordHandlingStrategy largeStreamRecordHandlingStrategy, boolean enableAggregation) {
-
-        if (StringUtils.isNotEmpty(region) && StringUtils.isNotEmpty(streamName)) {
+    public KinesisConfig(AbstractTargetModule tm) {
```

Comment:
We are changing signature of a public class, is there expected consequences for users/documentations?

Reply:
Yes it's changing the signature of a public class. I would argue it is acceptable in this case as this is a class that isn't expected to be used by external users. Anyone who is directly constructing `KinesisConfig` objects or extending this class is far outside of the intended use case of NeptuneExport. I wanted to remove the old constructor as the argument list has grown really unwieldy and difficult to maintain. There are a couple more constructors with 10+ arguments which I would like to try replacing in the future as well.

Perhaps a better choice for now is to retain the old constructor with an deprecated annotation. This would at the very least prevent us from needing to add additional arguments with every new feature.

Reply:
Yea I think adding deprecated annotation would be better and safer.

Reply:
Yeah, that makes sense. I have put in a deprecated constructor with the original signature